partner_firstname/tests/test_create.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2015 Grupo ESOC Ingeniería de Servicios, S.L. - Jairo Llopis.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+"""Test default values for models."""
+
+from odoo.tests.common import TransactionCase
+
+from .base import MailInstalled
+
+
+class PersonCase(TransactionCase):
+    """Test ``res.partner`` when it is a person."""
+
+    context = {"default_is_company": False}
+    model = "res.partner"
+
+    def setUp(self):
+        super(PersonCase, self).setUp()
+        self.good_values = {"firstname": "Núñez", "lastname": "Fernán"}
+        self.good_values["name"] = "{} {}".format(
+            self.good_values["firstname"], self.good_values["lastname"]
+        )
+        if "default_is_company" in self.context:
+            self.good_values["is_company"] = self.context["default_is_company"]
+        self.values = self.good_values.copy()
+
+    def tearDown(self):
+        # pylint: disable=W8121
+        self.record = (
+            self.env[self.model].with_context(self.context).create(self.values)
```

Comment:
Does it make any sense here replacing the whole context by itself?
I would remove with_context(self.context) more than disabling the pylint check

Reply:
I will try to remove the with_context and run the tests again. This came with the original source code by the way, although that doesn't mean we don't have to look into it. My bad.

Reply:
I did remove it and all tests continue passing